### PR TITLE
Automation of qhub init/render/deploy with Auth0 + Cloudflare

### PR DESCRIPTION
**Pull request by costrouc**

### CODE REVIEW

**Comment by costrouc** [comment]
This will create a way to automate the deployment of qhub.

Automation of dns and oauth requires the following environment variables
 - CLOUDFLARE_TOKEN
 - AUTH0_DOMAIN
 - AUTH0_CLIENT_ID
 - AUTH0_CLIENT_SECRET

# gcp example

qhub init gcp --domain gcp-test-1.qhub.dev --project qhub-test-gcp-1 --oauth-provider auth0 --oauth-auto-provision
qhub render -c qhub-config.yaml -f -o ./
qhub deploy -c qhub-config.yaml --dns-provider cloudflare --dns-auto-provision

This deployed in around 15 minutes
```

**Comment by tylerpotts** [comment]
I'll be using this to deploy on AWS and GCP today. Will comment back with my results

**Comment by costrouc** [comment]
I must stop at this point. Otherwise this PR will continue to grow. The initialization of a qhub cluster has been automated to the extent that it will not require any interaction with websites (github repo initialization, oauth, and dns).

```shell
qhub init gcp --domain do-1.qhub.dev --project qhub-test-gcp-1 --oauth-provider auth0 --oauth-auto-provision --repository github.com/quansight/qhub-test-deleteme --repository-auto-provision
qhub render -c qhub-config.yaml -f -o ./
qhub deploy -c qhub-config.yaml --dns-provider cloudflare --dns-auto-provision
git add -A
git commit -m "Initial Commit"
git push origin master
```

Notice how the git repo is now initialized and all github secrets are initailized.

**Comment by costrouc** [comment]
@aktech perfect thanks for adding that

File: qhub/provider/cicd/github.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import base64
+
+import requests
+from nacl import encoding, public
+
+
+def github_request(url, method="GET", json=None):
+    GITHUB_BASE_URL = "https://api.github.com/"
+
+    for name in ("GITHUB_USERNAME", "GITHUB_TOKEN"):
```
Comment (aktech) [inline qhub/provider/cicd/github.py:11]: Can we use `GH_USERNAME`, `GH_TOKEN`? As we cannot use these in Github Secrets, the usecase for puting these on Github secrets is for Integration testing.

Comment (costrouc) [reply qhub/provider/cicd/github.py:11]: Sounds good with me. Can we change the naming everywhere so that it is consistently `GH_USERNAME` and `GH_TOKEN`?

Comment (aktech) [reply qhub/provider/cicd/github.py:11]: Yeah, makes sense.

Comment (aktech) [reply qhub/provider/cicd/github.py:11]: I realise, this is not urgent as we can just map environment variables from github secret to a different key.

**Review by aktech — APPROVED** [review]
Looks Good to me.

**Pull request merged**